CHANGELOG.md

```diff
@@ -1,11 +1,29 @@
 # Release History
-## 0.10.0 (Unreleased)
+## 0.10.0 (2022-09-16)
 
 ### New Features:
+- Added support for evaluating Snowpark dataframes asynchronously :
+  - Added keyword argument `block` to most action APIs on Snowpark dataframes (which executes queries) to allow asynchronous evaluations.
+  - Added method `df.collect_nowait()` to asynchronous evaluations.
+  - Added class `AsyncJob` to retrieve results from asynchronously executed queries and check their statuses.
+  - This feature is experimental and APIs are subject to change.
 - Added support for `table_type` in `session.write_pandas()`. You can now choose from these `table_type` options: `"temporary"`, `"temp"`, and `"transient"`.
+- Added support for using Python structured data (`list`, `tuple` and `dict`) as literal values in Snowpark.
+- Added keyword argument `execute_as` to `functions.sproc()` and `session.sproc.register()` to allow registering a stored procedure as a caller or owner.
+- Added support for specifying a pre-configured file format when reading a stage file from Snowflake.
+
+### Breaking Changes:
+- When adding a package with a specific version for UDFs or stored procedures, if it is not supported by Snowflake, an exception will be raised immediately instead of when registering  or stored procedures.
+
+### Improvements:
+- Added support for displaying details of a Snowpark session when printing it out.
+
+### Bug Fixes:
+- Fixed a bug in which `df.copy_into_table()` and `df.write.save_as_table()` mistakenly created a new table if the table name is full-qualified and the table already exists.
 
 ### Deprecations:
-- Keyword Argument `create_temp_table` in `session.write_pandas()`.
+- Deprecated keyword argument `create_temp_table` in `session.write_pandas()`.
+- Deprecated the way of UDF invocation with arguments wrapped by a Python list or tuple.
```

Comment:
How about this:

"Deprecated invoking UDFs using arguments wrapped in a Python list or tuple."